File: CPP/coolakt/443. String Compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        if (n <= 1)
            return n;

        int writeIdx = 0;
        int count = 1;
        for (int i = 1; i <= n; i++) {
            if (i < n && chars[i] == chars[i - 1]) {
                count++;
            } else {
                chars[writeIdx++] = chars[i - 1];
                if (count > 1) {
                    string countStr = to_string(count);
                    for (char c : countStr) {
                        chars[writeIdx++] = c;
                    }
                }
                count = 1;
            }
        }
        return writeIdx;
    }
};

File: Java/Akash-212/443. String Compression.java
class Solution {
    public int compress(char[] chars) {
        if (chars == null || chars.length == 0) {
            return 0;
        }

        StringBuilder sb = new StringBuilder();

        int count = 1;

        for (int i = 0; i < chars.length; i++) {
            
            if (i + 1 < chars.length && chars[i] == chars[i + 1]) {
                count++;
            } else {
              
                sb.append(chars[i]);
            
                if (count > 1) {
                    sb.append(count);
                }
                
                count = 1;
            }
        }

        char[] compressedChars = sb.toString().toCharArray();
        System.arraycopy(compressedChars, 0, chars, 0, compressedChars.length);
        return compressedChars.length;
    }
}
